src/Lean/Meta/Tactic/Replace.lean

```diff
@@ -139,29 +140,64 @@ def _root_.Lean.MVarId.change (mvarId : MVarId) (targetNew : Expr) (checkDefEq :
 def change (mvarId : MVarId) (targetNew : Expr) (checkDefEq := true) : MetaM MVarId := mvarId.withContext do
   mvarId.change targetNew checkDefEq
 
+/-- Function to help do the revert/intro pattern, running some code inside a context
+where certain variables have been reverted before re-introing them.
+It will push `FVarId` alias information into info trees for you according to a simple protocol.
+
+- `fvarIds` is an array of `fvarIds` to revert. These are passed to
+  `Lean.MVarId.revert` with `preserveOrder := true`, hence the function
+  raises an error if they cannot be reverted in the provided order.
+- `k` is given the goal with all the variables reverted and
+  the array of reverted `FVarId`s, with the requested `FVarId`s at the beginning.
+  It must return a tuple of a value, an array describing which `FVarIds` to link,
+  and a mutated `MVarId`.
+
+The `a : Array (Option FVarId)` array returned by `k` is interpreted in the following way.
+The function will intro `a.size` variables, and then for each non-`none` entry we
+create an FVar alias between it and the corresponding `intro`ed variable.
+For example, having `k` return `fvars.map .some` causes all reverted variables to be
+`intro`ed and linked.
+
+Returns the value returned by `k` along with the resulting goal.
+ -/
```

Comment:
I think this docstring mixes too many separate concerns at one time. How about this?
```suggestion
/-- Runs the continuation `k` after temporarily reverting some variables from the local context of a metavariable (identified by `mvarId`), then reintroduces local variables as specified by `k`.

The argument `fvarIds` is an array of `fvarIds` to revert in the order specified. An error is thrown if they cannot be reverted in order.

Once the local variables have been reverted, `k` is passed `mvarId` along with an array of local variables that were reverted. This array always has `fvarIds` as a prefix, but it may contain additional variables that were reverted due to dependencies. `k` returns a value, a goal, an array of _link variables_.

Once `k` has completed, one variable is introduced for each link variable returned by `k`. If the returned variable is `none`, the variable is just introduced. If it is `some fv`, the variable is introduced and then linked as an alias of `fv` in the info tree. For example, having `k` return `fvars.map .some` as the link variables causes all reverted variables to be introduced and linked.

Returns the value returned by `k` along with the resulting goal.
 -/
```
I think this formulation separates the context manipulation from the UI concern of linking variables a bit more.